internal/transform/transform.go

```diff
@@ -425,20 +426,25 @@ func trimExtension(pathname string) string {
 	return pathname
 }
 
-func resolveIdForMatch(match *ImportMatch, opts *TransformOptions) string {
-	if strings.HasPrefix(match.Specifier, ".") && len(opts.Pathname) > 0 {
+func ResolveIdForMatch(id string, opts *TransformOptions) string {
+	// Try custom resolvePath if provided
+	if opts.ResolvePath != nil {
+		return opts.ResolvePath(id)
+	}
+	// Else use default resolvePath
```

Comment:
Comment on how we can remove the rest of this code in the future.

Reply:
Added in 4ccccec